src/index.ts

```diff
@@ -308,23 +321,29 @@ export class Paginator {
     });
 
     function makeRequest(query?: ParsedArguments|string) {
-      originalMethod(query, onResultSet);
+      if (parsedArguments.callback) {
+        originalMethod(query, onResultSet);
+      } else {
+        originalMethod(query).then(
+            // tslint:disable-next-line:no-any
+            (resp: any[]) => onResultSet(null, ...resp), onResultSet);
+      }
```

Comment:
why it needs this change? I think we can leave with this bcz we are supporting callback as well, when we are ready to make  all library async first then we can make this library as async first as well where we wouldnt need like this if condition.

Reply:
My thinking is as follows, if we are accepting `promise` based function, we should treat it as `promise` based internally as well. This will apply to the scenario when `originalMethod` is not `callbackified`.